### PR TITLE
Preserve DeepSeek sigmoid grouped routing metadata

### LINKED ISSUE #42426: [Bug]: Kimi-K2.6 intermittently outputs only "!!!!!!!!!!" in reasoning field with content null
### Your current environment
* Model: moonshotai/Kimi-K2.6 and redhatai/Kimi-K2.6-NVFP4
* Parsers: --reasoning-parser kimi_k2 --tool-call-parser kimi_k2
* vLLM image tested: v0.20.0 and v0.18.1
* GPUs: 8xB200

### Description

We are seeing intermittent bad generations from both `moonshotai/Kimi-K2.6` as well as `RedHatAI/Kimi-K2.6-NVFP4` served with vLLM. The response has `content: null`, while the reasoning field contains only repeated exclamation marks, e.g. `"!!!!!!!!!!"`.

This looks very similar to the Kimi-K2.5 issue discussed in #36763, especially this comment about FA4 NaNs at KV length >= 8192:
https://github.com/vllm-project/vllm/issues/36763#issuecomment-4170658649

The issue is not always immediate. In one deployment we saw it after the service had been running for a few days. We are still trying to isolate whether this is caused by FA4 MLA prefill, NVFP4 GEMM/MoE kernels, long-running state, or a specific traffic pattern.

But it seems like this appears when the model has either been running for a while and/or high traffic with large contexts.

When the model enters this bad state, requesting logprobs exposes NaNs in the response path. The request fails with:

```text
400 - {'error': {'message': 'Out of range float values are not JSON compliant: nan', ...}}
```
When:
```Bash
curl -X POST https://<HOST>/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer <TOKEN>" \
  -d '{
    "model": "kimi-k2.6",
    "messages": [
      {"role": "system", "content": "You are a helpful assistant."},
      {"role": "user", "content": "Hello, can you help me?"}
    ],
    "temperature": 0.7,
    "max_tokens": 20,
    "logprobs": true,
    "top_logprobs": 5
  }'
```

We are still trying to isolate whether this is caused by FA4 MLA prefill, FLASHINFER_MLA decode, NVFP4 GEMM/MoE kernels, long-running state, high-concurrency scheduling, or a specific traffic pattern.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by alexeldeib**
## Purpose

GLM-style DeepSeekV4/DSA models can use sigmoid routing with an expert-score correction bias, norm_topk_prob=True, n_group/topk_group metadata, and a non-unit routed scaling factor. With no group metadata passed to FusedMoE, the router is classified as RoutingMethodType.Unspecified: the bias-aware top-k router has no way to distinguish an explicit grouped DeepSeek router from not-a-grouped-router at all.

That classification matters for NVFP4 FlashInfer TRTLLM MoE. The monolithic TRTLLM expert path supports RoutingMethodType.DeepSeekV3, but intentionally does not claim Unspecified. Losing grouped routing metadata can therefore push the model onto a modular/pre-routed path instead of the monolithic from-logits path, bypassing FlashInfer fixes that are specific to DeepSeekV3 routing.

Propagate the model n_group and topk_group into FusedMoE for valid sigmoid routers only. A one-group router is semantically equivalent to selecting top-k over all experts, but it preserves the DeepSeekV3 routing contract for kernel selection and for backends that consume the routing metadata directly. Do not apply this to sqrtsoftplus DeepSeekV4 routing, which is a different routing method with different kernel support.

## Test Plan

Add CPU model tests that monkeypatch FusedMoE and verify valid sigmoid grouped routing metadata is preserved for both single-group and ordinary grouped cases. The tests also verify invalid group metadata is cleared and non-sigmoid routing still clears the same metadata.

## Test Result

new unit test cases passes, this PR combined with https://github.com/flashinfer-ai/flashinfer/pull/3875 also seems to resolve https://github.com/vllm-project/vllm/issues/42426#issuecomment-4908867521 in our real world usage

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by alexeldeib** [comment]
oh I can run _pre-commit_ not normal CI now nice
